Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -56,6 +56,25 @@ noncomputable def isPointwiseLeftKanExtensionLanUnit
     (LeftExtension.mk _ (L.lanUnit.app F)).IsPointwiseLeftKanExtension :=
   isPointwiseLeftKanExtensionOfIsLeftKanExtension (F := F) _ (L.lanUnit.app F)
 
+/-- If a left Kan extension is pointwise, then evaluating it at an object is isomorphic to
+taking a colimit. -/
+noncomputable def lanObjObjIsoColimit (F : C ⥤ H) [HasPointwiseLeftKanExtension L F] (X : D) :
+    (L.lan.obj F).obj X ≅ Limits.colimit (CostructuredArrow.proj L X ⋙ F) :=
+  let e₁ := isPointwiseLeftKanExtensionLanUnit L F
+  let e₂ := pointwiseLeftKanExtensionIsPointwiseLeftKanExtension L F
+  (Comma.rightIso (LeftExtension.isoOfIsPointwiseLeftKanExtension e₁ e₂)).app X
```

Comment:
```suggestion
  Limits.IsColimit.coconePointUniqueUpToIso
    (isPointwiseLeftKanExtensionLanUnit L F X)
    (Limits.colimit.isColimit (CostructuredArrow.proj L X ⋙ F))
```
(Actually, this could be generalized as a definition in the `IsPointwiseLeftKanExtensionAt` namespace.)

Most of the definitions in this PR are unnecessary/duplicates: the preferred API instead of `isoOfIsPointwiseLeftKanExtension` is `left/rightKanExtensionUnique` in the `.Basic` file.

Reply:
Can you clarify what you mean by "most of"? The two `isoOfIsPointwise...KanExtension` or also the simp lemmas?